beater/server.go

```diff
@@ -101,13 +101,20 @@ func createHandler(p processor.Processor, config Config, publish successCallback
 
 		// Limit size of request to prevent for example zip bombs
 		limitedReader := io.LimitReader(reader, config.MaxUnzippedSize)
-		err = p.Validate(limitedReader)
+
+		buf, err := ioutil.ReadAll(limitedReader)
```

Comment:
I remember there was a reason I passed the reader instead of the buffer to Validate, but I can't see why anymore.